code/modules/antagonists/infection/infection/commander.dm

```diff
@@ -0,0 +1,238 @@
+//Few global vars to track the infections
+GLOBAL_LIST_EMPTY(infections) //complete list of all infections made.
+GLOBAL_LIST_EMPTY(infection_nodes)
+GLOBAL_VAR(infection_core)
+GLOBAL_VAR(infection_commander)
+
+/mob/camera/commander
+	name = "Infection Commander"
+	real_name = "Infection Commander"
+	desc = "The commander. It controls the infection."
+	icon = 'icons/mob/cameramob.dmi'
+	icon_state = "marker"
+	mouse_opacity = MOUSE_OPACITY_ICON
+	move_on_shuttle = TRUE
+	see_in_dark = 8
+	invisibility = INVISIBILITY_OBSERVER
+	layer = FLY_LAYER
+	color = "#00a7ff"
+
+	pass_flags = PASSBLOB
+	faction = list(ROLE_INFECTION)
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	hud_type = /datum/hud/infection_commander
+	var/obj/structure/infection/core/infection_core = null // The infection commanders's core
+	var/obj/effect/meteor/infection/meteor = null // The infection's incoming meteor
+	var/infection_points = 0
+	var/max_infection_points = 300
+	var/upgrade_points = 0 // obtained by destroying beacons
+	var/all_upgrade_points = 0 // all upgrade points earned so far
+	var/last_attack = 0
+	var/list/infection_mobs = list()
+	var/list/resource_infection = list()
+	var/nodes_required = 1 //if the infection needs nodes to place resource and factory blobs
+	var/placed = FALSE
+	var/base_point_rate = 2 //for core placement
+	var/autoplace_time = 200 // a few seconds, just so it isnt sudden at game start
+	var/place_beacons_delay = 100
+	var/victory_in_progress = FALSE
+	var/infection_color = "#ffffff"
+	var/list/default_actions = list(/datum/action/innate/infection/creator/shield,
+									/datum/action/innate/infection/creator/resource,
+									/datum/action/innate/infection/creator/node,
+									/datum/action/innate/infection/creator/factory)
+	var/list/unlockable_actions = list(/datum/action/innate/infection/creator/turret)
+
+/mob/camera/commander/Initialize(mapload, starting_points = 0)
+	if(GLOB.infection_commander)
+		return INITIALIZE_HINT_QDEL // there can be only one
+	GLOB.infection_commander = src
+	infection_points = starting_points
+	autoplace_time += world.time
+	last_attack = world.time
+	if(infection_core)
+		infection_core.update_icon()
+	SSshuttle.registerHostileEnvironment(src)
+	addtimer(CALLBACK(src, .proc/generate_announcement), place_beacons_delay / 2)
+	addtimer(CALLBACK(src, .proc/place_beacons), place_beacons_delay)
+	for(var/type_action in default_actions)
+		var/datum/action/innate/infection/add_action = new type_action()
+		add_action.Grant(src)
+	.= ..()
```

Comment:
Why is this so late in initialize? None of this looks like it needs to be before parent inits?